sdk/storage/Azure.Storage.Blobs.ChangeFeed/src/ChangeFeedFactory.cs

```diff
@@ -0,0 +1,204 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs.ChangeFeed.Models;
+using Azure.Storage.Blobs.Models;
+
+namespace Azure.Storage.Blobs.ChangeFeed
+{
+    internal class ChangeFeedFactory
+    {
+        private readonly SegmentFactory _segmentFactory;
+        private readonly BlobContainerClient _containerClient;
+
+        public ChangeFeedFactory(
+            BlobServiceClient blobServiceClient)
+        {
+            _containerClient = blobServiceClient.GetBlobContainerClient(Constants.ChangeFeed.ChangeFeedContainerName);
+            _segmentFactory = new SegmentFactory(
+                _containerClient,
+                new ShardFactory(
+                    _containerClient,
+                    new ChunkFactory(
+                        _containerClient,
+                        new LazyLoadingBlobStreamFactory(),
+                        new AvroReaderFactory())));
+        }
+
+        public ChangeFeedFactory(
+            BlobContainerClient containerClient,
+            SegmentFactory segmentFactory)
+        {
+            _containerClient = containerClient;
+            _segmentFactory = segmentFactory;
+        }
+
+        public async Task<ChangeFeed> BuildChangeFeed(
+            bool async,
+            DateTimeOffset? startTime = default,
+            DateTimeOffset? endTime = default,
+            string continuation = default)
+        {
+            DateTimeOffset lastConsumable;
+            Queue<string> years = new Queue<string>();
+            Queue<string> segments = new Queue<string>();
+            ChangeFeedCursor cursor = null;
+
+            // Create cursor
+            if (continuation != null)
+            {
+                cursor = JsonSerializer.Deserialize<ChangeFeedCursor>(continuation);
+                ValidateCursor(_containerClient, cursor);
+                startTime = cursor.CurrentSegmentCursor.SegmentTime;
+                endTime = cursor.EndTime;
```

Comment:
Is it better if `endTime = minDate(cursor.EndTime, endTime);` to allow user specify a new endTime?

Reply:
Currently, I don't provide an overload for a user to provide both a cursor and end date.  I think it's conceptually simpler to have separate overloads for cursor and start/end time.